Loops in Cube::latLonRange need to use double instead of int

**ISIS version(s) affected**: 3.9.0

**Description**  
<!-- A clear and concise description of the problem. -->
The for-loops in Cube::latLonRange use an integer, but they need to iterate from 0.5 to sampleCount()+-.5 or lineCount()+0.5. By using an integer, they get truncated and iterate from 0 to sampleCount() or lineCount().

**How to reproduce**  
<!-- Code and/or config needed to reproduce the problem. -->
Clang gives a nice build warning when compiling on a Mac

**Possible Solution**  
<!--- Optional: only if you have suggestions on a fix/reason for the bug -->
Change to use a double.

**Additional context**  
<!-- Optional: any other context about the problem: log messages, screenshots, etc. -->
This method was added in #3402 

```
Building CXX object object...es/isis3.dir/base/objs/Cube/Cube.cpp.o
/work/users/jmapel/ISIS3/isis/src/base/objs/Cube/Cube.cpp:2184:23: warning: implicit conversion from 'double' to 'int' changes value from 0.5 to 0 [-Wliteral-conversion]
    for (int sample = 0.5; sample < sampleCount() + 0.5; sample++) {
             ~~~~~~   ^~~
/work/users/jmapel/ISIS3/isis/src/base/objs/Cube/Cube.cpp:2186:23: warning: implicit conversion from 'double' to 'int' changes value from 0.5 to 0 [-Wliteral-conversion]
      for (int line = 0.5; line < lineCount() + 0.5; line++) {
```

--- a/isis/src/base/objs/Cube/Cube.cpp
+++ b/isis/src/base/objs/Cube/Cube.cpp
@@ -2135,9 +2135,9 @@ namespace Isis {
 
 
 /**
- * Returns the latitude and longitude range for the Cube. More accurate than the minimum and 
- * maximum latitude and longitude from the mapping group. 
- * 
+ * Returns the latitude and longitude range for the Cube. More accurate than the minimum and
+ * maximum latitude and longitude from the mapping group.
+ *
  * @param[out] minLatitude minimum latitude present in the cube
  * @param[out] maxLatitude maximum latitude present in the cube
  * @param[out] minLongitude minimum longitude present in the cube
@@ -2181,16 +2181,16 @@ namespace Isis {
     maxLatitude = -99999;
     maxLongitude = -99999;
 
-    for (int sample = 0.5; sample < sampleCount() + 0.5; sample++) {
+    for (double sample = 0.5; sample < sampleCount() + 0.5; sample++) {
     // Checks to see if the point is in outer space
-      for (int line = 0.5; line < lineCount() + 0.5; line++) {
+      for (double line = 0.5; line < lineCount() + 0.5; line++) {
         if (useProj) {
           isGood = proj->SetWorld(sample, line);
         }
         else {
           isGood = cam->SetImage(sample, line);
         }
-      
+
         double lat, lon;
         if (isGood) {
           if (useProj) {
@@ -2198,10 +2198,10 @@ namespace Isis {
             lon = proj->UniversalLongitude();
           }
           else {
-            lat = cam->UniversalLatitude(); 
-            lon = cam->UniversalLongitude(); 
+            lat = cam->UniversalLatitude();
+            lon = cam->UniversalLongitude();
           }
-        
+
           // update mix/max lat/lons
           if (lat < minLatitude) {
             minLatitude = lat;
@@ -2209,23 +2209,23 @@ namespace Isis {
           else if (lat > maxLatitude) {
             maxLatitude = lat;
           }
-          
+
           if (lon < minLongitude) {
             minLongitude = lon;
           }
           else if (lon > maxLongitude) {
-            maxLongitude = lon; 
+            maxLongitude = lon;
           }
         }
       }
     }
-    if ( (minLatitude == 99999) || (minLongitude == 99999) || (maxLatitude == -99999) || 
+    if ( (minLatitude == 99999) || (minLongitude == 99999) || (maxLatitude == -99999) ||
     (maxLongitude == -99999) ) {
       QString msg = "Unable to calculate a minimum or maximum latitutde or longitude.";
         throw IException(IException::Unknown, msg, _FILEINFO_);
     }
   }
-  
+
   /**
    * Write the Pvl labels to the cube's label file. Excess data in the attached
    *   labels is set to 0.

